lib/tasks/export.rake

```
@@ -0,0 +1,161 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+require 'zip'
+require 'fileutils'
+
+# This pollutes the global namespace
+def append_paper_metadata_header(csv)
+  csv << ["doi", "title", "status", "authors", "academic_editors", "handling_editors", "cover_editors", "reviewers"]
+end
+
+def append_paper_metadata(csv, paper)
+  authors = paper.authors.map(&:full_name).join(",")
+  academic_editors = paper.academic_editors.map(&:full_name).join(",")
+  handling_editors = paper.handling_editors.map(&:full_name).join(",")
+  cover_editors = paper.cover_editors.map(&:full_name).join(",")
+  reviewers = ReviewerReport.where(task_id: paper.tasks.pluck(:id))
+                .map(&:user).uniq.compact.map(&:full_name).join(",")
+  csv << [paper.doi,
+          paper.title,
+          paper.publishing_state,
+          authors,
+          academic_editors,
+          handling_editors,
+          cover_editors,
+          reviewers]
+end
+
+def mk_zip_entry(zos, entry_name, time=nil)
+  dt = if time.nil?
+         nil
+       else
+         Zip::DOSTime.parse(time.utc.strftime("%Y-%m-%d %H:%M:%S"))
+       end
+  entry = Zip::Entry.new(zos, entry_name, nil, nil, nil, nil, nil, nil, dt)
+  zos.put_next_entry(entry)
+  yield
+end
+
+def zip_add_url(zos, name, url)
+  return if url.nil?
+  conn = Faraday.new
+  resp = conn.get(url)
+  last_modified = if resp.headers['Last-Modified']
+                    DateTime.rfc2822(resp.headers['Last-Modified'])
+                  else
+                    nil
+                  end
+  mk_zip_entry(zos, name, last_modified) do
+    zos << resp.body
+  end
+rescue Faraday::ConnectionFailed => ex
+  puts(ex)
```

should we fail harder here?

---

I can't remember why I did this. Yes, we should remove the `rescue`

---

fixed.